static/app/utils/useHasMounted.tsx

```diff
@@ -0,0 +1,14 @@
+import {useEffect, useRef} from 'react';
+/**
+ * Provides a boolean indicating if the component has completed its initial mount.
+ * @returns true if the component has mounted, false otherwise.
+ */
+export function useHasMounted() {
```

Comment:
I decided to introduce this hook because of issues I had in tests but I am happy to push this code in a separate PR if needed

Reply:
I think we can delete this in favour of `useIsMountedRef`?